### ISSUE TITLE
Cannot compile on Ubuntu 14.04

### ISSUE THREAD

**Issue by nico87** [google/flatui#2]
Running `cmake -G'Unix Makefiles' .` inside the flatui main folder in Ubuntu 14.04 causes the following error:

```
The source directory

    /home/claudio/flatui/dependencies/mathfu

  does not contain a CMakeLists.txt file.
```

All the needed libraries are installed. I've also tried to checkout the entire project again bu I had the very same issue.

**Comment by lilianmoraru** [comment]
I can confirm the same issue on Ubuntu 15.10. I installed the suggested dependencies listed below the error.
After this message, you see this error:
```
~/Git/flatui/dependencies/harfbuzz ~/Git/flatui
$HOME/Git/flatui/cmake/harfbuzz/autogen_if_changed.sh: line 32: $HOME/Git/flatui/dependencies/harfbuzz/autogen.sh: No such file or directory
CMake Error at cmake/harfbuzz/CMakeLists.txt:46 (message):
  CMake encountered an error while configuring harfbuzz.
```
Note: I replaced my actual home folder with `$HOME`

**Comment by lilianmoraru** [comment]
@nico87 There is no setup script so seems like you have to go into the `dependencies` folder and do `git clone` for every project you have there.
Example:
```
git clone https://github.com/google/flatui.git
cd flatui/dependencies

git clone https://github.com/behdad/harfbuzz.git
git clone https://github.com/google/mathfu.git
git clone https://github.com/google/fplbase.git
...
```

**Comment by hak** [comment]
Hello,
It's using git submodules.
Could you try
git clone --recursive https://github.com/google/flatui
?

hak

**Comment by lilianmoraru** [comment]
@hak Yes, it worked.

**Comment by nico87** [comment]
@hak and @lilianmoraru It worked. Thanks! Maybe it would be good to add the checkout instructions in the readme file.

On my system both flatuisample and flatuitest work correctly.

**Comment by gwvo** [comment]
That's an SDL error, and I'm not quite sure what could cause it. Anything non-standard about your system? Are you in a VM? Do other applications that use OpenGL work?

**Comment by nico87** [comment]
@lilianmoraru I've tested the sample and test builds on a pc with an old ATI X1900 graphics adapter and I get the same error "The specified window has not been made current". I think it's about OpenGL. With this adapter I've never been able to run an OpenGL application under Linux.
Maybe you are in the same situation?

**Comment by lilianmoraru** [comment]
@gwvo @nico87 Yes, I ran the application in a VM. VMWare 12 with the tools installed.
Yes, other applications that use OpenGL work.
For example, we have Qt Quick 2 applications running in the VM. Qt Quick 2 has a hard requirement on OpenGL 2.1 or higher.

**Comment by gwvo** [comment]
FlatUI (and FPLBase which it uses for rendering) currently asks for OpenGL 3.1 on Linux. This is because of instancing support, which FlatUI doesn't yet use, so it could be downgraded to 2.1.

To test if this is the case, change 3 to 2 in dependencies/fplbase/src/renderer.cpp:114

**Comment by lilianmoraru** [comment]
@gwvo It looks fuzzy and weird but it works.

**Comment by gwvo** [comment]
Yes, our current test isn't pretty, if that's what you're referring to :)

I'll see what I can do to lower the minimum requirements to 2.1
